compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEOps.td

```diff
@@ -355,29 +355,60 @@ def AMDAIE_ChannelOp: AMDAIE_Op<"channel", [
 
     ```mlir
     %tile = amdaie.tile(%c0, %c0)
-    %channel = amdaie.channel(%tile, 0, port_type = DMA)
+    %channel = amdaie.channel(%tile, 0, port_type = DMA, direction = MM2S)
     ```
   }];
 
   let arguments = (
     ins Index:$tile,
         ConfinedAttr<I8Attr, [IntMinValue<0>]>:$value,
-        StrmSwPortTypeAttr:$port_type
+        StrmSwPortTypeAttr:$port_type,
+        DMAChannelDir:$direction
   );
 
   let extraClassDeclaration = [{
     TileOp getTileOp();
   }];
 
   let assemblyFormat = [{ 
-    `(` $tile `,` $value `,` `port_type` `=` $port_type `)` attr-dict
+    `(` 
+    $tile `,`
+    $value `,` 
+    `port_type` `=` $port_type `,` 
+    `direction` `=` $direction
+    `)`
+    attr-dict
   }];
 }
 
 //===----------------------------------------------------------------------===//
 // IREE AMDAIE Npu Ops
 //===----------------------------------------------------------------------===//
 
+def AMDAIE_NpuAddressPatchOp: AMDAIE_Op<"npu.address_patch"> {
+  let summary = "Operation to patch the address inside a buffer descriptor";
+  let description = [{
+    This NPU controller operation patches the address inside the buffer
+    descriptor with provided ID on the specified column. This enables codegen to
+    provide an argument index and offset at compile time, which is then
```

Comment:
sorry I haven't been paying attention: arg index and offset to what? hal.buffer bindings or something?